### PR TITLE
Fix: with no `SERVER_PASSWORD` can still apply setting

### PR DESCRIPTION
**Pull request by win5923**
## Choices
*  Allow setting `ServerPassword` to an empty string through env `SERVER_PASSWORD`
* Resolve the issue of applying settings without manually modifying PalWorldSettings.ini when there is no server password.
* Addressing a similar issue as described in https://github.com/thijsvanloef/palworld-server-docker/issues/275#issue-2115930445.

## Test instructions

1. Comment out the SERVER_PASSWORD or set it to an empty string.
2. Open the game and check if the `ServerPassword` in PalWorldSettings.ini is set to an empty string.

## Checklist before requesting a review

* [x] I have performed a self-review of my code
* [x] I've added documentation about this change to the README.
* [x] I've not introduced breaking changes.

### CODE REVIEW

**Comment by win5923** [comment]
Do I need to apply this setting to `ADMIN_PASSWORD` as well?

**Comment by Dashboy1998** [comment]
> Do I need to apply this setting to `ADMIN_PASSWORD` as well?

[rcon-cli](https://github.com/gorcon/rcon-cli/issues/37) doesn't work with an empty password. 
RCON protocol itself refuses all requests when no password is set.

If there is no admin password then you aren't able to become an admin: `/adminpassword ""`

File: scripts/start.sh [hunk 1]
```diff
@@ -122,6 +122,9 @@ if [ -n "${SERVER_PASSWORD}" ]; then
     SERVER_PASSWORD=$(sed -e 's/^"\(.*\)"$/\1/' -e "s/^'\(.*\)'$/\1/" <<< "$SERVER_PASSWORD")
     echo "SERVER_PASSWORD=${SERVER_PASSWORD}"
     sed -E -i "s/ServerPassword=\"[^\"]*\"/ServerPassword=\"$SERVER_PASSWORD\"/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
+else
+    echo "No SERVER_PASSWORD"
+    sed -E -i "s/ServerPassword=\"[^\"]*\"/ServerPassword=\"\"/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
```
Comment (Dashboy1998) [inline scripts/start.sh:127]:
```suggestion
elif [ -v SERVER_PASSWORD ]; then
    echo "No SERVER_PASSWORD"
    sed -E -i "s/ServerPassword=\"[^\"]*\"/ServerPassword=\"\"/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
```

I think we should remove SERVER_PASSWORD from the Dockerfile. 
If by default we have no variable named SERVER_PASSWORD then nothing will be overridden unless you explicitly state no password (With the elif -v). 

This way we don't always overwrite the password. I'm not worried about the extra write so much as removing a password someone has set in their file but not set as an environmental variable.

Comment (win5923) [reply scripts/start.sh]: I also think this way is better. Thanks for your suggestion.

**Comment by thijsvanloef** [comment]
@win5923 Will review tonight!

**Review by thijsvanloef — APPROVED** [review]
made some changes, but ready to merge

**Pull request merged**